loadable_apps/Makefile

```diff
@@ -53,53 +52,50 @@
 -include $(TOPDIR)/.config
 -include $(TOPDIR)/Make.defs
 
-ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
-BUILD_SUBDIRS = micomapp wifiapp
+BUILDIRS   := $(dir $(wildcard */Make.defs))
+OUTPUT_BIN_DIR = $(TOPDIR)/../build/output/bin
+CONFIGURED =
 
-USER_BIN_DIR = $(TOPDIR)/../build/output/bin
+define ADD_APPS
+	include $(1)Make.defs
+endef
+$(foreach BDIR, $(BUILDIRS), $(eval $(call ADD_APPS,$(BDIR))))
 
 define DIR_template
 $(1)_$(2):
-	$(Q) $(MAKE) -C $(1) $(3) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" USER_BIN_DIR="$(USER_BIN_DIR)" CROSSDEV=$(CROSSDEV)
+	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" USER_BIN_DIR="$(OUTPUT_BIN_DIR)" CROSSDEV=$(CROSSDEV)
 endef
+$(foreach BDIR, $(CONFIGURED), $(eval $(call DIR_template,$(BDIR),context)))
+$(foreach BDIR, $(CONFIGURED), $(eval $(call DIR_template,$(BDIR),all)))
+$(foreach BDIR, $(BUILDIRS), $(eval $(call DIR_template,$(BDIR),preconfig)))
+$(foreach CDIR, $(BUILDIRS), $(eval $(call DIR_template,$(CDIR),clean)))
+$(foreach CDIR, $(BUILDIRS), $(eval $(call DIR_template,$(CDIR),distclean)))
 
-all: build verify install
-.PHONY: all build clean install distclean verify
-
-$(foreach DIR, $(BUILD_SUBDIRS), $(eval $(call DIR_template,$(DIR),build, all)))
-$(foreach DIR, $(BUILD_SUBDIRS), $(eval $(call DIR_template,$(DIR),clean,clean)))
-$(foreach DIR, $(BUILD_SUBDIRS), $(eval $(call DIR_template,$(DIR),install,install)))
-$(foreach DIR, $(BUILD_SUBDIRS), $(eval $(call DIR_template,$(DIR),verify,verify)))
-
-# Build program(s) in each sub-directory
-
-build: $(foreach DIR, $(BUILD_SUBDIRS), $(DIR)_build)
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
+$(foreach BDIR, $(CONFIGURED), $(eval $(call DIR_template,$(BDIR),verify)))
+$(foreach BDIR, $(CONFIGURED), $(eval $(call DIR_template,$(BDIR),install)))
 
-# Install each program in the romfs directory
+all: build verify install
+.PHONY: all build clean install distclean depend verify preconfig
 
-install: $(foreach DIR, $(BUILD_SUBDIRS), $(DIR)_install)
+install: $(foreach BDIR, $(CONFIGURED), $(BDIR)_install)
 
-# Verify each program for undefined symbols
+verify: $(foreach BDIR, $(CONFIGURED), $(BDIR)_verify)
+else # !CONFIG_APP_BINARY_SEPARATION
 
-verify: $(foreach DIR, $(BUILD_SUBDIRS), $(DIR)_verify)
-else
-all:
+all: build
+.PHONY: all build clean distclean depend preconfig
+endif # CONFIG_APP_BINARY_SEPARATION
```

Comment:
Do we need to do anything in the #else part?
As I understand, the contents of the loadable_apps must be built only if app separation is enabled.

Reply:
@kishore-sn As I told on chat, I don't want to use this folder only for separated application. When app separation is enabled, it has to be separated binary. But with another build ways, it has to be a part of binary. It is more convenient for user. For this, we had better rename the folder name with another PR.